.NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseDatePeriodExtractor.cs

```diff
@@ -682,5 +726,63 @@ private bool IsDateRelativeToNowOrToday(ExtractResult er)
 
             return false;
         }
+
+        // Matches "within (the next)?" part (in beforeStr or afterStr) in "within Days/Weeks/Months/Years"
+        private void MatchWithinNextPrexixRegex(string text, Token duration, out int startToken, out int endToken, out bool matchSuccess, bool inPrefix)
+        {
+            var beforeStr = text.Substring(0, duration.Start);
+            var afterStr = text.Substring(duration.Start + duration.Length);
+            startToken = -1;
+            endToken = -1;
+            matchSuccess = false;
+            var match = inPrefix ? config.WithinNextPrefixRegex.MatchEnd(beforeStr, trim: true) : config.WithinNextPrefixRegex.MatchBegin(afterStr, trim: true);
+            if (match.Success)
+            {
+                startToken = inPrefix ? match.Index : duration.Start;
+                endToken = inPrefix ? duration.End : duration.End + match.Index + match.Length;
+                var durationStr = text.Substring(duration.Start, duration.Length);
+                var matchDate = config.DateUnitRegex.Match(durationStr);
+                var matchTime = config.TimeUnitRegex.Match(durationStr);
+
+                matchSuccess = matchDate.Success && !matchTime.Success;
+
+                if (!inPrefix)
+                {
+                    // Check prefix for "next"
+                    match = config.FutureRegex.MatchEnd(beforeStr, trim: true);
+                    if (match.Success)
+                    {
+                        startToken = match.Index;
+                    }
+
+                }
+            }
+        }
+
+        private List<Token> ExtractWithinNextPrefix(string subStr, bool inPrefix, ExtractResult extractionResult)
```

Comment:
This is just basically an ExtractToMethod refactor, correct?

Reply:
yes, as you suggested in the Date PR, it is used to avoid repeating the same piece of code for beforeStr and afterStr.